iron-overlay-behavior.html

```diff
@@ -397,20 +396,10 @@
     },
 
     _onCaptureClick: function(event) {
-      // attempt to close asynchronously and prevent the close of a tap event is immediately heard
-      // on target. This is because in shadow dom due to event retargetting event.target is not
-      // useful.
-      if (!this.noCancelOnOutsideClick && (this._manager.currentOverlay() == this)) {
-        this._cancelJob = this.async(function() {
-          this.cancel();
-        }, 10);
-      }
-    },
-
-    _onClick: function(event) {
-      if (this._cancelJob) {
-        this.cancelAsync(this._cancelJob);
-        this._cancelJob = null;
+      if (Polymer.dom(event).path.indexOf(this) === -1 &&
```

Comment:
This is a minor detail, but it would probably be ideal to check the event path last. I think computing the event path is relatively most expensive of the three booleans.